### PR TITLE
imap: fix imap list subscribe

### PR DESCRIPTION
**Pull request by sileht**
imap: Attempt to fix subscription issue.

### CODE REVIEW

**Comment by gahr** [comment]
Can you please merge 7f6a78 into this branch? It fixes subscribe/unsubscribe from the mailboxes list menu.

**Comment by sileht** [comment]
This introduces a crash when I switch account. I will continue to work on it.

File: imap/imap.c [hunk 1]
```diff
@@ -404,27 +404,24 @@ int imap_prepare_mailbox(struct Mailbox *m, struct ImapMbox *mx,
...
   struct ImapMbox mx;
-  struct MailboxNode *np = NULL;
 
-  memset(&mx, 0, sizeof(mx));
+  if (imap_parse_path(path, &mx) < 0)
+    return -1;
 
-  STAILQ_FOREACH(np, &AllMailboxes, entries)
+  *adata = imap_ac_data_find(&mx);
+  if (!adata)
```
Comment (sileht) [inline imap/imap.c]: arf, wrong pointer

**Comment by sileht** [comment]
It's fixed now

**Pull request merged**